### PR TITLE
feat: add outline (document outline) PDFs option

### PR DESCRIPTION
**Pull request by bungeman**
**What kind of change does this PR introduce?**

feature

**Did you add tests for your changes?**

Yes

**If relevant, did you update the documentation?**

Yes

**Summary**

Chromium [introduced generateDocumentOutline](https://chromium.googlesource.com/chromium/src.git/+/9c2c6e785b9c644887c466ae49d7947e6c690171) in 121. This allows for a PDF document outline (sometimes known as bookmarks) to be generated from the html document outline. This CL adds a `outline` property to request this extra PDF output.

**Does this PR introduce a breaking change?**

No

**Other information**

This change is heavily based on https://github.com/puppeteer/puppeteer/pull/11182 .

### CODE REVIEW

**Review by OrKoN — CHANGES_REQUESTED** [review]
Thanks for the PR! it looks like there are some build issues. Make sure `npm run build`, `npm run format` and `npm run docs` all pass and submit any changes produced by the formatter or the doc generator.

**Comment by bungeman** [comment]
Hmmm... it looks like `generateDocumentOutline` ended up in devtools-protocol [r1226504](https://github.com/ChromeDevTools/devtools-protocol/commit/9a978928587a2ba68d479fbfc0be735802a5450d) but it looks like devtools-protocol is still pinned to `r1213968`. So this will need to wait for a devtools-protocol roll. Guessing based on timing it looks like that may happen soon?

File: test/src/page.spec.ts [hunk 1]
```diff
@@ -2015,6 +2015,25 @@ describe('Page', function () {
       }
     });
 
+    it('can print to PDF with outline', async () => {
```
Comment (OrKoN) [inline test/src/page.spec.ts:2018]: btw, let's move this test to the cdp subfolder

**Comment by Lightning00Blade** [comment]
@bungeman the change for Chrome 121 (https://github.com/puppeteer/puppeteer/pull/11743) landed would you please rebase and address the comments.

**Pull request closed without merging**